verification/verification_test.go

```diff
@@ -98,15 +101,36 @@ func GetTestTarget(support_needed []string) (TestDPEInstance, error) {
 
 // Get the emulator target
 func GetEmulatorTarget(support_needed []string) (TestDPEInstance, error) {
+	// TODO : Get the supported modes from emulator and then check.
+	var instance TestDPEInstance = &DpeEmulator{exe_path: *target_exe}
+	if instance.HasPowerControl() {
+		err := instance.PowerOn()
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer instance.PowerOff()
+	}
+
+	client, err := NewClient384(instance)
+	if err != nil {
+		return nil, errors.New("Error in getting client")
+	}
+
+	rsp, err := client.GetProfile()
+	if err != nil {
+		return nil, errors.New("Unable to get profile")
+	}
 
-	value := reflect.ValueOf(DpeEmulator{}.supports)
+	support := Support{}
+
+	value := reflect.ValueOf(support.ToSupport(rsp.Flags))
 	for i := 0; i < len(support_needed); i++ {
 		support := reflect.Indirect(value).FieldByName(support_needed[i])
 		if !support.Bool() {
 			return nil, errors.New("Error in creating dpe instances - supported feature is not enabled in emulator")
 		}
 	}
-	var instance TestDPEInstance = &DpeEmulator{exe_path: *target_exe}
+	instance.SetSupport(support)
```

Comment:
`Support` cannot be set on the emulator. Either the target supports what is needed or it doesn't. So I think this line can be removed and then the SetSupport target API can be removed.